### PR TITLE
handle floating promises in /inter-protocol and /vats

### PR DESCRIPTION
**Pull request by turadg**
refs: #6000

## Description

Because we don't have the floating promise lint in CI, they creep back. The last time inter-protocol was cleared:
- https://github.com/Agoric/agoric-sdk/pull/6803

This clears them again. To do so it reworks a few promise handling flows. Those are in separate commits from the mechanical ones.

This is stacked upon https://github.com/Agoric/agoric-sdk/pull/7752 because that changes shortfall reporting in a way that was relevant to the test changes here.

This ~has~ omits changes that are related to auction fixes:
- https://github.com/Agoric/agoric-sdk/pull/7748

### Security Considerations

n/a
### Scaling Considerations

n/a
### Documentation Considerations

n/a

### Testing Considerations

Revised tests. The vaults-upgrade bootstrapTest had to shrink in scope.

### CODE REVIEW

File: packages/inter-protocol/src/vaultFactory/liquidation.js [hunk 1]
```diff
@@ -96,7 +96,7 @@ const scheduleLiquidationWakeups = async (
   if (TimeMath.compareAbs(now, nominalStart) > 0) {
     if (TimeMath.compareAbs(now, endTime) < 0) {
       if (cancelToken) {
-        E(timer).cancel(cancelToken);
+        void E(timer).cancel(cancelToken);
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/vaultFactory/liquidation.js:99]: If we don't wait, I think we should assign a new token so the old one doesn't get used to soon.
```
cancelToken = makeCancelToken();
```

File: packages/inter-protocol/src/auction/scheduler.js [hunk 2]
```diff
@@ -176,7 +177,7 @@ export const makeScheduler = async (
               'Unable to start auction cleanly. skipping this auction round.',
             ),
           );
-          finishAuctionRound();
+          await finishAuctionRound();
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/auction/scheduler.js:180]: Why?  `finishAuctionRound()` isn't *async*.

File: packages/inter-protocol/src/auction/scheduler.js [hunk 3]
```diff
@@ -188,15 +189,15 @@ export const makeScheduler = async (
       case 'before':
         break;
       case 'during':
-        advanceRound();
+        await advanceRound();
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/src/auction/scheduler.js:192]: nor is `advanceRound()`.

File: packages/inter-protocol/test/auction/test-scheduler.js [hunk 4]
```diff
@@ -1080,15 +1079,9 @@ test('schedule anomalies', async t => {
   const veryLateStart = baseTime + 5n * oneCycle;
   await timer.advanceTo(veryLateStart);
 
-  // This schedule is published as a side effect of closing out the incomplete
-  // auction. The next one follows immediately and is correct.
```
Comment (Chris-Hibbert) [inline packages/inter-protocol/test/auction/test-scheduler.js:1084]: Also removed in another pending PR.

**Review by Chris-Hibbert — CHANGES_REQUESTED** [review]

File: packages/vats/src/core/basic-behaviors.js [hunk 5]
```diff
@@ -489,7 +489,8 @@ export const makeAddressNameHubs = async ({
 
   const perAddress = address => {
     const { nameHub, myAddressNameAdmin } = makeMyAddressNameAdminKit(address);
-    myAddressNameAdmin.reserve(WalletName.depositFacet);
+    // This may not finish before another reserve() call, but this one will win
+    void myAddressNameAdmin.reserve(WalletName.depositFacet);
```
Comment (dckc) [inline packages/vats/src/core/basic-behaviors.js:493]: `makeMyAddressNameAdminKit` is deprecated, yes? I'm not confident how this solo client stuff should work. I'd be inclined to have ci continue to gripe about it. Or leave an `XXX` or something.

File: packages/vats/src/core/utils.js [hunk 6]
```diff
@@ -305,7 +305,8 @@ export const makeMyAddressNameAdminKit = address => {
     getMyAddress: () => address,
   });
   // reserve space for deposit facet
-  myAddressNameAdmin.reserve(WalletName.depositFacet);
+  // This may not finish before another reserve() call, but this one will win
+  void myAddressNameAdmin.reserve(WalletName.depositFacet);
```
Comment (dckc) [inline packages/vats/src/core/utils.js:309]: Again, I'm not confident about this.

File: packages/vats/src/virtual-purse.js [hunk 7]
```diff
@@ -162,7 +162,7 @@ const prepareVirtualPurseKit = zone =>
       minter: {
         /** @type {Retain} */
         async retain(payment, optAmountShape) {
-          this.state.mint || Fail`minter cannot retain without a mint.`;
+          !!this.state.mint || Fail`minter cannot retain without a mint.`;
```
Comment (dckc) [inline packages/vats/src/virtual-purse.js:165]: :+1:

**Review by Chris-Hibbert — APPROVED** [review]
As discussed, if the changes for auctions and liquidation are moved to a different PR, I'm fine with what remains.  We can discuss them there.

**Pull request merged**